redash/models/__init__.py

```diff
@@ -206,6 +210,17 @@ def delete(self):
         db.session.commit()
         return res
 
+    def save_schema(self, schema_info):
```

Comment:
Let's make this a classmethod since you're not really acting on a DataSource instance but generally updating data source related metadata. It's better to keep it away from objects.